lib/p2p/Pool.ts

```diff
@@ -96,26 +107,11 @@ class Pool extends EventEmitter {
     }
 
     if (this.server) {
-      let externalIp: string | undefined;
-      // Fetch the external IP if no address was specified by the user
-      if (this.addresses.length === 0) {
-        try {
-          externalIp = await getExternalIp();
-
-          this.logger.info(`retrieved external IP: ${externalIp}`);
-        } catch (error) {
-          this.logger.error(error.message);
-        }
-      }
-
       await this.listen();
       this.bindServer();
 
-      if (externalIp) {
-        this.addresses.push({
-          host: externalIp,
-          port: this.listenPort!,
-        });
+      if (this.config.detectExternalIp) {
```

Comment:
The new method doesn't keep the `if (this.addresses.length === 0)` check before detecting external ip. We should probably keep that, since if we've specified a listening address then we don't need to automatically detect one as well I figure. I think this line would make the most sense to have it, so `this.config.detectexternalip && this.addresses.length === 0`.

Reply:
I thought, we put that configuration option to get rid of it since the config is false it doesn't matter to have an address or not to detect external ip. Putting it back.

Reply:
I see these two options working together. The user can provide listening addresses and can also ask to detect external addresses.

Reply:
The idea of the external ip detection originally was to provide a listening address when none is provided. I suppose we could change that, but then at a minimum we'd want to check to make sure that it's not duplicating a value that's already specified in the listening addresses.

Reply:
adding a duplicate check then before adding the external ip to the addresses